identity-iota/src/credential/validator.rs

```diff
@@ -19,28 +30,29 @@ use crate::error::Result;
 use crate::tangle::Client;
 use crate::tangle::TangleResolve;
 
-#[derive(Clone, Debug, PartialEq, Serialize)]
-pub struct CredentialValidation<T = Object> {
-  pub credential: Credential<T>,
-  pub issuer: DocumentValidation,
-  pub subjects: BTreeMap<String, DocumentValidation>,
-  pub verified: bool,
+// used internally during credential validation
+#[allow(clippy::large_enum_variant)] // The smaller variants occur at most once each
+enum CredentialDeficiencyEvent<'a> {
+  // the Url corresponds to the url of the deactivated document
+  DeactivatedSubjectDocument((&'a Url, DocumentValidation<Deactivated>)),
+  Expired(Timestamp),
+  Dormant(Timestamp),
```

Comment:
Looks like those timestamps aren't really used? The deficiency handler just converts it into the `CredentialDeficiency` anyway without looking at the timestamp.